.github/workflows/build-upstream-nightly.yml

```diff
@@ -1,19 +1,45 @@
-name: Build Upstream Nightly Packages
+name: Build & check Upstream Nightly
 
 on:
   schedule:
     # * is a special character in YAML so you have to quote this string
     - cron:  '20 10 * * *'
   workflow_dispatch:
-    
+    inputs:
+      composer_cache_dir:
+        description: A key to version the composer cache. Can be incremented if you need to bust the cache.
+        required: false
+        default: ''
 
 jobs:
   deploy: 
-    uses: ./.github/workflows/build-preview-mirror.yml
+    uses: ./.github/workflows/deploy.yml
     with:
       repo: https://upstream-nightly.mage-os.org
-      remote_dir: '/var/www/upstream-nightly.mage-os.org/html/'
-      entrypoint: 'src/make/upstream-nightly.js'
+      remote_dir: /var/www/upstream-nightly.mage-os.org/html/
+      entrypoint: src/make/upstream-nightly.js
+      delete: false
     secrets:
       SERVER_SSH_KEY: ${{ secrets.SERVER_SSH_KEY }}
       REMOTE_USER: ${{ secrets.REMOTE_USER }}
+  compute-nightly-service-versions:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.supported-version.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: graycoreio/github-actions-magento2/supported-version@allow_using_next_version
+        with:
+          kind: nightly
+        id: supported-version
+  integration-check:
+    needs: [deploy, compute-nightly-service-versions]
+    uses: graycoreio/github-actions-magento2/.github/workflows/integration.yaml@main
+    with:
+      package_name: mageos/demo-package
```

Comment:
This needs to be `[mage-os/demo-package](https://packagist.org/packages/mage-os/demo-package)